src/jit/emitxarch.cpp

```diff
@@ -4812,6 +4962,24 @@ void emitter::emitIns_SIMD_R_R_R(instruction ins, regNumber reg, regNumber reg1,
         emitIns_R_R(ins, emitTypeSize(simdtype), reg, reg2);
     }
 }
+
+void emitter::emitIns_SIMD_R_R_R_I(
+    instruction ins, regNumber reg, regNumber reg1, regNumber reg2, ssize_t ival, var_types simdtype)
+{
+    if (UseVEXEncoding() && reg1 != reg)
+    {
+
+        emitIns_R_R_R_I(ins, emitTypeSize(simdtype), reg, reg1, reg2, (int)ival);
+    }
+    else
+    {
+        if (reg1 != reg)
+        {
+            emitIns_R_R_I(INS_movaps, emitTypeSize(simdtype), reg, reg1, (int)ival);
```

Comment:
Don't think this is correct. `movaps` is a `R_R` instruction

Reply:
You are right, will fix by dropping whole if statement. Thanks

Reply:
I don't think dropping the if statement is correct.

We still have, a target register, two operands, and an immediate, which isn't encodable on non-VEX machines if `reg1 != reg`